dotCMS/src/main/java/com/dotcms/api/system/event/SystemEventsFactory.java

```diff
@@ -161,10 +161,8 @@ private void push(final SystemEvent systemEvent, boolean forceNewTransaction) th
 				Logger.error(this, msg, e);
 				throw new DotDataException(msg, e);
 			} finally {
-
-				if (localTransaction) {
-
-					DbConnectionFactory.closeSilently();
+				if (localTransaction && isNewConnection) {
+					HibernateUtil.closeSessionSilently();
```

Comment:
@jcastro-dotcms @jdotcms shouldn't we use the annotation approach here instead?

Reply:
This change was suggested directly by @jdotcms

Reply:
The thing is there is some scenario where the transaction is force and it is not supported by the annotation